mmpose/datasets/pipelines/shared_transform.py

```diff
@@ -27,11 +28,16 @@ class ToTensor:
         results (dict): contain all information about training.
     """
 
+    def __init__(self, device='cpu'):
+        self.device = device
+
     def __call__(self, results):
+        to_tensor = (lambda x: torch.from_numpy(x.astype('float32')).permute(
```

Comment:
It seems that when I used **multiple GPUs** and this [config](https://github.com/open-mmlab/mmpose/blob/master/configs/body/2d_kpt_sview_rgb_img/topdown_heatmap/coco/hrnet_w32_coco_256x192.py) for **testing**, the AP is near **zero**, which is not reasonable.

But when I used **single GPU**, the result is consistent . 

I also tried theses two cases:
1. Only keep your modification about in the `ToTensor`, the result with single GPU is correct while result in multiple GPUs is still near **zero**. 
2. Only keep your modification about in the `NormalizeTensor`, the results on single and multiple GPUa are both correct. 

I was wondering if it have something to do with the modifications you add inside `ToTensor`, which may cause some in place modification to the data implicitly.

Reply:
I didn't test in multiple GPU before. It seems unreasonable that the modification won't work in multiple GPU. What's the test file and command did you use? I will then check it and find the way out.

Reply:
**UPDATE**: Previously I mistook the running path of MMPose. Now the results of single GPU and multiple GPUs are the same, but both got mAP near **zero**.

I use this config https://github.com/open-mmlab/mmpose/blob/master/configs/body/2d_kpt_sview_rgb_img/topdown_heatmap/coco/hrnet_w32_coco_256x192.py.  and download the corresponding checkpoint from this link https://download.openmmlab.com/mmpose/top_down/hrnet/hrnet_w32_coco_256x192-c78dce93_20200708.pth

1. For single GPU testing, I run:
```
PYTHONPATH="$(dirname $0)/..":$PYTHONPATH python tools/test.py configs/body/2d_kpt_sview_rgb_img/topdown_heatmap/coco/hrnet_w32_coco_256x192.py checkpoints/hrnet_w32_coco_256x192-c78dce93_20200708.pth
```

2. For multiple GPUs testing, I run:
```
GPUS=8 GPUS_PER_NODE=8 CPUS_PER_TASK=12 SRUN_ARGS='--quotatype=auto' ./tools/slurm_test.sh mm_human test_pr1320 configs/body/2d_kpt_sview_rgb_img/topdown_heatmap/coco/hrnet_w32_coco_256x192.py checkpoints/hrnet_w32_coco_256x192-c78dce93_20200708.pth
```

Another interesting phenomena is that when I run multi-GPU training using the same config and this command:
```
GPUS=8 GPUS_PER_NODE=8 CPUS_PER_TASK=12 SRUN_ARGS='--quotatype=auto' ./tools/slurm_train.sh mm_human train_pr1320 configs/body/2d_kpt_sview_rgb_img/topdown_heatmap/coco/hrnet_w32_coco_256x192.py work_dirs/
```
The evaluation results during training **looked just fine**., with mAP near 0.65 after 10 epochs. 

Considering the above testing results, this is a little bit strange.

Reply:
When I change your modification in `ToTensor` back to `F.to_tensor`, the mAP is correct. 

So maybe it was the `to_tensor` function you add that somehow changes the data inplace.